### PR TITLE
ci: add auto release workflows

### PR DESCRIPTION
**Pull request by Xunzhuo**
**What type of PR is this?**

**What this PR does / why we need it**:

Sub of #9

**Which issue(s) this PR fixes**:

- [ ] Release Workflow: 
  - [ ] support automatcally build and publish i2gw multi-arch binaries when pushing a new tag.

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by Xunzhuo** [comment]
xref: https://github.com/kubernetes-sigs/ingress2gateway/issues/21

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Copyright © 2022 Kubernetes Authors
```
Comment (robscott) [inline .github/workflows/release.yaml:1]:
```suggestion
# Copyright © 2023 Kubernetes Authors
```

**Review by robscott — COMMENTED** [review]
Thanks for working on this @Xunzhuo!

How can we test this?

File: .github/workflows/release.yaml (hunk 1 above)
Comment (Xunzhuo) [reply .github/workflows/release.yaml]: Hey @robscott,I think we can test it in the fork repo, after the previous PRs get merged refer to https://github.com/kubernetes-sigs/ingress2gateway/issues/21#issuecomment-1484915839.

**Comment by robscott** [comment]
@Xunzhuo let us know if you want to follow up on these, for now I'm going to close out in favor of more recent PRs.

**Pull request closed without merging**